CASMNET-1580:  Update CAN documentation to reflect BICAN

johren-hpe opened the pull request:
## Summary and Scope

There are many pages in the documentation that talk about CAN.   These needed to be updated to now talk about CMN and CHN as well.   Most, if not all, of these pages have not been through the link, spelling, and style checks yet so they also include changes to satisfy those checks.

Removed operations/network/customer_access_network/Required_Labels_if_CAN_is_Not_Configured.md because this workaround no longer works.  This worked when we had the site network connected to w001 but now that it is connected to m001, we will not be able to run the pods that look for this label on the master nodes.   The only pod I found that uses this label is cray-keycloak. 

Removed operations/network/external_dns/Add_NCNs_and_UANs_to_External_DNS.md because powerdns now creates names for the NCNs and UANs.  Therefore, this is no longer necessary.

@mtupitsyn also asked that the PNG extensions be made lower case so you will see several of those files moved and links corrected for that.

## Issues and Related PRs

* Resolves [CASMNET-1580](https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1580)
* Resolves [CASMINST-4701](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4701)

## Testing

All markdown changes. 


## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

johren-hpe commented:
The shell_check error is a known issue that is being discussed at https://cray.slack.com/archives/C01F7HW8LBX/p1654278746322669.   Waiting to hear the result of that discussion.

rustydb commented on operations/system_layout_service/Add_UAN_CAN_IP_Addresses_to_SLS.md:
````diff
@@ -73,26 +74,27 @@ This procedure requires administrative privileges.
 
-    ```
+    ```bash
     ncn-m001# nslookup uan01-can.can
````
Do `uan01-can.can` and `uan01-can` exist intentionally? Do we need to have `uan01-can.can` in the document at all?

johren-hpe replied:
I want to check with Jeff to see if this procedure is still needed.   But I see Jeff is on leave so I left it in for now.

rustydb commented on operations/network/customer_accessible_networks/Troubleshoot_CMN_Issues.md:
````diff
@@ -25,59 +26,60 @@ Example output:
 
-```screen
+```bash
 ncn-w002# ip route | grep default
````
nit: make this copy-paste friendly, remove `ncn-w002#` since this command is agnostic to ncn.

johren-hpe replied:
Addressed

rustydb commented:
Since this PR touches so many networking docs, should we update `CODEOWNERS` via this PR to include `@Cray-HPE/management-network` for at least `operations/network/`?

nrockershousen commented:
The /operations/index.md file should be updated since files are being removed/added.

Entries to remove:

- MetalLB_in_BGP-Mode_Configuration.md
- Add_NCNs_and_UANs_to_External_DNS.md
- Connect_to_the_CAN.md
- Customer_Access_Network_CAN.md
- Externally_Exposed_Services.md
- Required_Labels_if_CAN_is_Not_Configured.md

johren-hpe commented:
/backport release/1.2

rustydb commented on operations/network/external_dns/Ingress_Routing.md:
````diff
@@ -1,27 +1,27 @@
+For example, the configuration below controls the ingress routing for `prometheus.cmn.SYSTEM_DOMAIN_NAME`:
 
 ```bash
 ncn-w001# kubectl get vs -n sysmgmt-health cray-sysmgmt-health-prometheus
````
This might be extra work, if so just ignore/dismiss. Could you please also fix-up the prompts in this page to be more copy-paste friendly? This should move the `ncn-w001#` prompt out of all the code snippets so that this is more copy-paste friendly. https://github.com/Cray-HPE/docs-csm/blob/MTL-1695/introduction/documentation_conventions.md#step-examples

johren-hpe replied:
I need to keep the scope of this ticket limited to the changes required for the ticket and the changes required to pass the lint checkers.

johren-hpe replied:
@rustydb you already made this change in your MTL-1695 PR.   That has now been rebased in.

johren-hpe commented on operations/network/customer_accessible_networks/bi-can_arista_metallb_peering.md:
````diff
@@ -105,9 +105,9 @@ SLS entries from the above output:
                 "Name": "chn-switch-2"
 ```
````
I will fix this after I sort out all of the new conflicts.

johren-hpe replied:
Done

johren-hpe commented:
@spillerc-hpe @rustydb 
* pulled in MTL-1695 changes
* resolved a zillion conflicts
* fixed new linting issues that were pulled in
* addressed all comments

rustydb commented:
> @spillerc-hpe @rustydb
> 
> * pulled in [MTL-1695](https://jira-pro.its.hpecorp.net:8443/secure/QuickSearch.jspa?searchString=MTL-1695) changes
> * resolved a zillion conflicts
> * fixed new linting issues that were pulled in
> * addressed all comments

I don't think the CODEOWNERS _needs_ leading backslashes but they don't hurt to have: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

rustydb commented on operations/network/metallb_bgp/MetalLB_Configuration.md:
````diff
@@ -0,0 +1,111 @@
+The CAN or CHN configuration is set in the `csi config init` input:
+
+```bash
+linux# csi config init
````
Remove the prompt for consistency with the other `csi config init` command above.

rustydb reviewed (approved):
Approving, I just have my outstanding comment.

rustydb commented:
@johren-hpe I resolved the conflict for `CODEOWNERS` for you.
